File: 3533-snake-in-matrix/snake-in-matrix.py
class Solution:
    def finalPositionOfSnake(self, n: int, commands: List[str]) -> int:
        row, col = 0, 0

        for command in commands:
            if command == 'UP':
                row -= 1
            elif command == 'DOWN':
                row += 1
            elif command == 'LEFT':
                col -= 1
            elif command == 'RIGHT':
                col += 1
            
        final_position = row * n + col
        return final_position

File: 3533-snake-in-matrix/snake-in-matrix.js
/**
 * @param {number} n
 * @param {string[]} commands
 * @return {number}
 */
var finalPositionOfSnake = function(n, commands) {
    /* let idx = 0;
    for (const cmd of commands) {
        if (cmd === 'RIGHT') idx += 1;
        else if (cmd === 'LEFT') idx -= 1;
        else if (cmd === 'DOWN') idx += n;
        else if (cmd === 'UP') idx -= n;
    }
    return idx; */

    let row = 0, col = 0;

    for (const cmd of commands) {
        if (cmd === 'RIGHT') col += 1;
        else if (cmd === 'LEFT') col -= 1;
        else if (cmd === 'DOWN') row += 1;
        else if (cmd === 'UP') row -= 1;
    }
    return row * n + col;
};